tests/lib/rules/array-type.js

```diff
@@ -9,114 +10,796 @@
+            options: ["generic"],
+            errors: [
+                {
+                    messageId: "errorStringGeneric",
+                    line: 2,
+                    column: 15,
+                },
+            ],
+        },
+        {
+            code: `interface ArrayClass<T> {
+    foo: Array<T>;
+    bar: T[];
+    baz: Arr<T>;
+}`,
+            output: `interface ArrayClass<T> {
+    foo: Array<T>;
+    bar: Array<T>;
+    baz: Arr<T>;
+}`,
+            options: ["generic"],
+            errors: [
+                {
+                    messageId: "errorStringGeneric",
+                    line: 3,
+                    column: 10,
+                },
+            ],
+        },
+        {
+            code: `function barFunction(bar: ArrayClass<String>[]) {
+    return bar.map(e => e.bar);
+}`,
+            output: `function barFunction(bar: Array<ArrayClass<String>>) {
+    return bar.map(e => e.bar);
+}`,
+            options: ["generic"],
+            errors: [
+                {
+                    messageId: "errorStringGeneric",
+                    line: 1,
+                    column: 27,
+                },
+            ],
+        },
+        {
+            code: `let barVar: ((c: number) => number)[];`,
+            output: `let barVar: Array<(c: number) => number>;`,
+            options: ["generic"],
+            errors: [
+                {
+                    messageId: "errorStringGeneric",
+                    line: 1,
+                    column: 13,
+                },
+            ],
+        },
+        {
+            code: `type barUnion = (string|number|boolean)[];`,
+            output: `type barUnion = Array<string|number|boolean>;`,
+            options: ["generic"],
+            errors: [
+                {
+                    messageId: "errorStringGeneric",
+                    line: 1,
+                    column: 17,
+                },
+            ],
+        },
+        {
+            code: `type barIntersection = (string & number)[];`,
+            output: `type barIntersection = Array<string & number>;`,
             options: ["generic"],
             errors: [
                 {
-                    message: genericMessage,
+                    messageId: "errorStringGeneric",
                     line: 1,
-                    column: 8
+                    column: 24,
+                },
+            ],
+        },
+        {
+            code: `interface FooInterface {
+    '.bar': {baz: string[];};
+}`,
+            output: `interface FooInterface {
+    '.bar': {baz: Array<string>;};
+}`,
+            options: ["generic"],
+            errors: [
+                {
+                    messageId: "errorStringGeneric",
+                    line: 2,
+                    column: 19,
+                },
+            ],
+        },
+    ],
+});
+
+// eslint rule tester is not working with multi-pass
+// https://github.com/eslint/eslint/issues/11187
```

Comment:
for time being i added additional tests outside of rule-tester from eslint to check if output is correct for nested cases